profiles/source_profile.go

```diff
@@ -429,12 +431,80 @@ func NewSourceProfileConnection(source string, params map[string]string) (Source
 	return conn, nil
 }
 
+type DirectConnectionConfig struct {
+	Host     string `json:"host"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	Port     string `json:"port"`
+	DbName   string `json:"dbName"`
+}
+
+type DatastreamConnProfile struct {
+	Name     string `json:"name"`
+	Location string `json:"location"`
+}
+
+type DataflowConfig struct {
+	Location      string `json:"location"`
+	Network       string `json:"network"`
+	Subnetwork    string `json:"subnetwork"`
+	HostProjectId string `json:"hostProjectId"`
+}
+
+type DataShard struct {
+	DataShardId           string                `json:"dataShardId"`
+	SrcConnectionProfile  DatastreamConnProfile `json:"srcConnectionProfile"`
+	DestConnectionProfile DatastreamConnProfile `json:"destConnectionProfile"`
+	DataflowConfig        DataflowConfig        `json:"dataflowConfig"`
+	TmpDir                string                `json:"tmpDir"`
+	StreamLocation        string                `json:"streamLocation"`
+	LogicalShards         []LogicalShard        `json:"databases"`
+}
+
+type LogicalShard struct {
+	DbName         string   `json:"dbName"`
+	TableInclude   []string `json:"tableInclude"`
+	TableExclude   []string `json:"tableExclude"`
+	LogicalShardId string   `json:"databaseId"`
+	RefDataShardId string   `json:"refDataShardId"`
+}
+
+type ShardConfigurationDataflow struct {
+	SchemaShard DirectConnectionConfig `json:"schemaShard"`
+	DataShards  []*DataShard   `json:"dataShards"`
+}
+
+type ShardConfigurationBulk struct {
+	SchemaShard DirectConnectionConfig   `json:"schemaShard"`
+	DataShards  []DirectConnectionConfig `json:"dataShards"`
+}
+
+type ShardConfigurationDMS struct {
```

Comment:
add TODO

Reply:
Added TODO here and in other places as well.